src/main/java/com/redhat/exhort/integration/backend/ExhortIntegration.java

```diff
@@ -126,8 +133,46 @@ public void configure() {
       .setProperty(Constants.API_VERSION_PROPERTY, constant(Constants.API_VERSION_V4))
       .to(direct("analysis"));
 
+    from(direct("v4batchAnalysis"))
+      .routeId("v4BatchAnalysis")
+      .to(direct("batchAnalysis"));
+
     from(direct("analysis"))
       .routeId("dependencyAnalysis")
+      .to(direct("preProcessAnalysisRequest"))
+      .process(this::processAnalysisRequest)
+      .to(direct("analyzeSbom"))
+      .to(direct("report"))
+      .to(direct("postProcessAnalysisRequest"));
+
+    from(direct("batchAnalysis"))
+      .routeId("batchDependencyAnalysis")
+      .to(direct("preProcessAnalysisRequest"))
+      .process(this::processBatchAnalysisRequest)
+      .to(direct("analyzeSboms"))
+      .to(direct("batchReport"))
+      .to(direct("postProcessAnalysisRequest"));
+
+    from(direct("analyzeSbom"))
+      .routeId("analyzeSbom")
+      .enrich(direct("getTrustedContent"), tcResponseAggregation)
+      .to(direct("findVulnerabilities"))
+      .transform().method(ProviderAggregationStrategy.class, "toReport");
+
+    from(direct("analyzeSboms"))
+      .routeId("analyzeSboms")
+      .split(body(), new GroupedBodyAggregationStrategy())
+        .parallelProcessing()
+          .setProperty(Constants.ROOT_REF_PROPERTY, bodyAs(DependencyTree.class).method("rootRef"))
+          .setProperty(Constants.DEPENDENCY_TREE_PROPERTY, body())
+          .to(direct("analyzeSbom"))
+          .bean(RootTransformer.class, "attachUBIRecommendation")
```

Comment:
In my opinion, all the recommendations should be added in the TrustedContentIntegration because there is where we add recommendations and remediations. This should be done only if there are no recommendations for a given package. The `TcResponseHandler` would be a good place to do that.
This way even an external SBOM submitted by another client (e.g. trustification) will receive these recommendations.